scala-thoughts.md

```diff
@@ -0,0 +1,45 @@
+##Scala Thoughts
+
+This was my first time programming in Scala, and it definitely took some time 
+and practice to get the hang of it. For one, I only have minimal experience
+with functional programming; the only functional language I can think of that 
+I've written substantial programs in is with SML. Compared to this experience, 
+Scala felt rather overwhelming because there are so many different ways to call
+and write functions. I think the earlier functions that I wrote in this lab
+tended to be more inefficient because I wrote them in a more imperative manner. 
+But it was hard to not write in an imperative style because Scala didn't 
+explicitly stop me from doing so. It took a lot of reading through 
+documentation and looking at sample code to start to get the hang of a more 
+Scala-y, functional style of programming.
+
+When working with Scala Collections, I was a little initimidated but also
+impressed by how many different methods there were to modify and perform 
+operations on lists. The only ones that I really learned how to use were 
+fold, map, and filter, but I found so many more while looking through the 
+documentation, such as zip, scan, and reduce. Many times, after writing the 
+body of a function, I realized that there was a much more elegant way of 
+implementing it because there are so many built-in methods. It was somehow 
+both annoying and satisfying to see that 10 lines of code could sometimes be 
+reduced to a single line using a powerful method like fold. 
```

Comment:
This is probably due to the quality of the problems we were working with. Functional programming is certainly powerful, but it's not only this straightforward.